src/main/kotlin/com/codecraft/ai/api/engine/KeyFusionAPI.kt

```diff
@@ -0,0 +1,68 @@
+package com.codecraft.ai.api.engine
+
+import com.codecraft.ai.api.models.Text2ImageRequest
+import com.codecraft.ai.api.models.Text2ImageResponse
+import io.ktor.client.*
+import io.ktor.client.call.*
+import io.ktor.client.engine.cio.*
+import io.ktor.client.plugins.*
+import io.ktor.client.plugins.contentnegotiation.*
+import io.ktor.client.request.*
+import io.ktor.http.*
+import io.ktor.serialization.kotlinx.json.*
+import io.ktor.server.application.*
+import io.ktor.server.response.*
+import kotlinx.serialization.json.Json
+import java.nio.charset.StandardCharsets
+import java.util.*
+
+suspend fun startText2ImageProcess(call: ApplicationCall) {
+    call.request.apply {
+        val promptFromUser = queryParameters[QueryParam.PROMPT.strRepresentation] ?: return Error.EMPTY_PROMPT.let {
+            call.respond(
+                HttpStatusCode(
+                    it.code,
+                    it.strRepresentation
+                )
+            )
+        }
+
+        connectToSDAndGetImage(
+            Text2ImageRequest(
+                prompt = promptFromUser,
+                steps = queryParameters[QueryParam.STEPS.strRepresentation]?.toInt() ?: 25,
+                width = queryParameters[QueryParam.WIDTH.strRepresentation]?.toInt() ?: 512,
+                height = queryParameters[QueryParam.HEIGHT.strRepresentation]?.toInt() ?: 512,
+            ),
+        )?.let {
```

Comment:
what if `connectToSDAndGetImage` returns null? I would expect we should always call `call.respond(it)`

Reply:
respond parameter can not be nullable, so in case of null I will send error code